File: 1768-merge-strings-alternately/1768-merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
  const length = Math.max(word1.length, word2.length);
  const result = [];
  
  for (let i = 0; i < length; i++) {
    if (word1[i]) {
      result.push(word1[i]);
    }
    
    if (word2[i]) {
      result.push(word2[i]);
    }
  }
  
  return result.join("");
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        len1 = len(word1)
        len2 = len(word2)
        result = ""

        for i in range(0, max(len1, len2)):
            if i < len1:
                result += word1[i]
            if i < len2:
                result += word2[i]

        return result
